doschema/jsvalidation.py

```diff
@@ -0,0 +1,326 @@
+
+        self.uri = None
+        self.resolver = None
+
+    def validate_schema(self, curr_schema, curr_field):
+        """Go through the schema and retrieve schema's particular fields.
+
+        :param curr_schema: Schema or subschema that is currently processed.
+        :param curr_field: Tuple with path to currently processed field.
+        """
+        self.validate_collection(curr_schema, curr_field)
+
+        while '$ref' in curr_schema:
+            path = curr_schema.get('$ref')
+            curr_schema = self.resolver.resolve(path)[1]
+
+        field = self.fields_types_dict.get(curr_field)
+
+        if "type" in curr_schema:
+            if field and curr_schema['type'] != field.field_type:
+                err_msg = "{0} type mismatch in schemas {1} and {2}."
+                raise JSONSchemaCompatibilityException(
+                    field.schema_index,
+                    self.uri,
+                    err_msg.format(
+                        self.make_json_pointer(curr_field),
+                        field.schema_index,
+                        self.uri
+                    )
+                )
+            self.add_field_to_dict(curr_field, curr_schema['type'])
+
+            if curr_schema['type'] == 'array':
+                self.validate_array(
+                    curr_schema['items'],
+                    curr_field + ('items',)
+                )
+
+        if "enum" in curr_schema:
+            enum_type = self.validate_enum(
+                curr_schema['enum'],
+                curr_field,
+                curr_schema.get('type')
+            )
+            if field is not None and field.field_type != enum_type:
+                err_msg = "Enum {0} conflicting itself error in schema {1}."
+                raise JSONSchemaEnumException(
+                    self.uri,
+                    err_msg.format(
+                        self.make_json_pointer(curr_field),
+                        self.uri
+                    )
+                )
+            self.add_field_to_dict(curr_field, enum_type)
+
+        if 'properties' in curr_schema:
+            if field and field.field_type != "object":
+                err_msg = "Root {0} is not object error in schema {1}"
+                raise JSONSchemaRootNotObjectException(
+                    self.uri,
+                    err_msg.format(
+                        self.make_json_pointer(curr_field),
+                        self.uri
+                    )
+                )
+
+            self.add_field_to_dict(curr_field, 'object')
+            for prop in curr_schema['properties']:
+                self.validate_schema(
+                    curr_schema['properties'][prop],
+                    curr_field + (prop,)
+                )
+
+        if "dependencies" in curr_schema:
+            if field and field.field_type != "object":
+                err_msg = "Dependencies {0} error in schema {1}."
+                raise JSONSchemaDependenciesException(
```

Comment:
Same as above. It also looks like a conflict between two types. A message like this might give more information:
`"Conflicting type for field '...' previously found in '...' with type '...', found next in '...' where "dependencies" imply an "object" type."`